cogito/putter.go

```diff
@@ -262,23 +259,24 @@ func checkGitRepoDir(dir, owner, repo string) error {
 	if err != nil {
 		return fmt.Errorf(".git/config: remote: %w", err)
 	}
-	left := []string{"github.com", owner, repo}
```

Comment:
So as discussed, for this PR we will not validate the domain, but I still want to give the endpoint information to the end user, just in case. So even though we don't strictly compare the `gu.URL.Host` with the configured github_api_endpoint we at least print both. So at least this will simplify the debugging a bit. 

For me this is still an open question, but we will discuss it when @marco-m-pix4d is back. For now I go with this beacuse:
- It its how the authors did it in the first commit
- @marco-m-pix4d recommended to ease the validation for now: https://github.com/Pix4D/cogito/pull/142#discussion_r1364192679

Reply:
I see only two options here: 
- either we ease the validation as its done here
- or we support configuring both `github_endpoint` and `github_api_endpoint`. 

How would this work?
- if both are unset in source config: `github_endpoint` points to: `https://github.com` and `github_api_endpoint` points to: `https://api.github.com`
- if only `github_endpoint` is set.  `github_api_endpoint` points to: `https://<github_endpoint>/api/v3/`. This is the default API endpoint for GitHub Enterprise in my understanding
- if both are set we use the values that are set

This would enable us to due this validation using the configured: `github_endpoint` and do the API request to `github_api_endpoint`.

@wanderanimrod Opinions on the alternative solution?

Reply:
@marco-m-pix4d see the 3 possibilities:
https://github.com/google/go-github/blob/66ed84deba8936684a4a6bb09b00135f326064fc/github/github.go#L363

and the naming convention they use.

Reply:
@aliculPix4D thanks for the explanation.

Looking at the official documentation for GH Enterprise (https://docs.github.com/en/enterprise-server@3.10/rest/overview/resources-in-the-rest-api?apiVersion=2022-11-28), I could not find, as you mentioned, the fact that if the hostname contains the `api` subdomain, then the URL should not contain "/api/v3": it seems that all GH Enterprise installation want (or support) hostname/api/v3 ...

I then though that instead of considering `go-github` as authoritative (it comes from Google, not GibHub), we could maybe look at the `gh` tool (which does come from GitHub). There I found https://github.com/cli/cli/blob/trunk/internal/ghinstance/host.go, which seems to contain all the logic to make a decision. Have a look also at the test file, https://github.com/cli/cli/blob/20baccfa85d15963eb1ab4f750da0da37b0af7f5/internal/ghinstance/host_test.go.

There, in function `RESTPrefix` https://github.com/cli/cli/blob/20baccfa85d15963eb1ab4f750da0da37b0af7f5/internal/ghinstance/host.go#L85-L96, they seem to have all the logic:
```go
func RESTPrefix(hostname string) string {
	if isGarage(hostname) {
		return fmt.Sprintf("https://%s/api/v3/", hostname)
	}
	if IsEnterprise(hostname) {
		return fmt.Sprintf("https://%s/api/v3/", hostname)
	}
	if strings.EqualFold(hostname, localhost) {
		return fmt.Sprintf("http://api.%s/", hostname)
	}
	return fmt.Sprintf("https://api.%s/", hostname)
}
```
from that, it seems that there are only the following possibilities:
```
garage.github.com                    -> garage.github.com/api/v3/
anything not github.com (enterprise) -> anything/api/v3/
github.localhost                     -> api.github.localhost
github.com                           -> api.github.com (this I think)
```

Now, as we know, anything below `internal` means that it can break at any moment, but it still seems reasonable to me. It is similar to the logic in `go-github`. Maybe we could lift the whole file and be done with it (license is MIT) ?

Going back to your proposal:

> or we support configuring both `github_endpoint` and `github_api_endpoint`.

Do I understand correctly that the reason to introduce 2 parameters is to be able to use directly `github_endpoint` to compare with the git remote from `.git/config` without having to remove the `/api/v3` path prefix?

The reason I am going slowly is to see if we can both keep the configuration as simple as possible and avoid painting ourselves in a corner if we make an error.

Reply:
> Do I understand correctly that the reason to introduce 2 parameters is to be able to use directly github_endpoint to compare with the git remote from .git/config without having to remove the /api/v3 path prefix?

Yes, but given your findings that is not needed.

As I see now, we should support setting only: `github_hostname`, and default is: `github.com`, and then we should add the `api.` prefix for default or `/api/v3` suffix to the `github_hostname` when API url is required.

This will allow for simple validation  against the git remote from .git/config and using the correct API url when required.